scripts/o3de/o3de/project_manager_interface.py

```diff
@@ -268,18 +269,30 @@ def get_gem_info(gem_path: str) -> dict or None:
     return dict()
 
 
-def get_all_gem_infos(project_path: str) -> list:
+def get_all_gem_infos(project_path: pathlib.Path or None) -> list:
     """
-        Get list of all avaliable gem paths for project at project_path using get_all_gems
-
-        Gather together list of dicts for each gem using get_gem_json_data
+        Get list of all avaliable gem json data for a project. If no project
+        path is provided, the gems for this engine are returned.
 
         :param project_path: Project path to gather avaliable gem infos for
 
-        :return list of dicts containing gem infos.
+        :return list of dicts containing gem json data
     """
-    return list()
+    all_gem_json_data = manifest.get_gems_json_data_by_name(project_path=project_path,
+                                                            include_engine_gems=True,
+                                                            include_manifest_gems=True)
+
+    # flatten into a single list
+    all_gem_json_data = [gem_json_data for gem_versions in all_gem_json_data.values() for gem_json_data in gem_versions]
+
+    # it's easier to determine which gems are engine gems here rather than in c++
+    # because this gem might belong to a different engine than the one Project Manager is
+    # running out of
+    engine_path = manifest.get_project_engine_path(project_path=project_path) if project_path else manifest.get_this_engine_path() 
+    for i, gem_json_data in enumerate(all_gem_json_data):
+        all_gem_json_data[i]['engine_gem'] = engine_path in gem_json_data['path'].parents
```

Comment:
Can't this check against the `manifest.get_engine_gems` gems?
```suggestion
    # it's easier to determine which gems are engine gems here rather than in c++
    # because this gem might belong to a different engine than the one Project Manager is
    # running out of
    engine_path = manifest.get_project_engine_path(project_path=project_path) if project_path else manifest.get_this_engine_path() 
    engine_gems = manifest.get_engine_gems(engine_path);
    for i, gem_json_data in enumerate(all_gem_json_data):
        all_gem_json_data[i]['engine_gem'] = gem_json_data['path'] in engine_gems
```

Reply:
I found at least one case where the drive letter case was different leading to invalid matches e.g. 'D:' vs 'd:' because the 'path' values in `gem_json_data` are strings and so are those in `engine_gems`. I think this is due to how you can have an `engine_path` entry in a `user/project.json` that points to an all lower case path where as the manifest.json may have an upper case version.
```
>>> import pathlib
>>> p = pathlib.Path('c:/Windows')
>>> p.as_posix()
'c:/Windows'

>>> p2 = pathlib.Path('C:/Windows')
>>> p2.as_posix()
'C:/Windows'

>>> p3 = pathlib.Path('c:/windows')
>>> p3.as_posix()
'c:/windows'
```

Reply:
That is why the pathlib.Path should be used for comparison.
Doing a string compare is incorrect and a path compare should occur.